### PR TITLE
Add global variables to the debug hit message

### PR DESCRIPTION
**Pull request by pranavan15**
## Purpose
Currently, debug hit messages only contain the details of the local variables. This PR adds the global variable support to the debug messages.

### CODE REVIEW

**Comment by hasithaa** [comment]
@pranavan15 Can we add a test case for this?

File: tests/ballerina-unit-test/src/test/resources/test-src/debugger/test_variables.bal [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    int count,
+    string last,
+};
+
+type Person object {
+public int age,
+public string name,
+public Person? parent,
+private string email = "default@abc.com",
+};
```
Comment (hasithaa) [inline tests/ballerina-unit-test/src/test/resources/test-src/debugger/test_variables.bal]: add eol.

Comment (pranavan15) [reply tests/ballerina-unit-test/src/test/resources/test-src/debugger/test_variables.bal]: Added.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java [hunk 2]
```diff
@@ -342,6 +346,44 @@ private MessageDTO generateDebugHitMessage(WorkerExecutionContext ctx, LineNumbe
...
                 callingLine.getLineNumber());
         message.addFrame(frameDTO);
+
+        int pkgIndex = ctx.callableUnitInfo.getPackageInfo().pkgIndex;
+
+        PackageVarInfo[] packageVarInfoEntries = ctx.programFile.getPackageInfo(ctx.programFile.getEntryPkgName())
+                .getPackageInfoEntries();
+        for (PackageVarInfo packVarInfo : packageVarInfoEntries) {
+            if (!packVarInfo.getName().startsWith(META_DATA_VAR_PATTERN)) {
+                VariableDTO variableDTO = new VariableDTO(packVarInfo.getName(), "Global");
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java]: Shall we make "Global" a constant?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java [hunk 3]
```diff
@@ -342,6 +346,44 @@ private MessageDTO generateDebugHitMessage(WorkerExecutionContext ctx, LineNumbe
         FrameDTO frameDTO = new FrameDTO(pck, functionName, callingLine.getFileName(),
                 callingLine.getLineNumber());
         message.addFrame(frameDTO);
+
+        int pkgIndex = ctx.callableUnitInfo.getPackageInfo().pkgIndex;
+
+        PackageVarInfo[] packageVarInfoEntries = ctx.programFile.getPackageInfo(ctx.programFile.getEntryPkgName())
+                .getPackageInfoEntries();
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java]: We generally break new line with period (.) being on the same line and the rest will be moved to next line.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java [hunk 4]
```diff
@@ -342,6 +346,44 @@ private MessageDTO generateDebugHitMessage(WorkerExecutionContext ctx, LineNumbe
...
+        for (PackageVarInfo packVarInfo : packageVarInfoEntries) {
+            if (!packVarInfo.getName().startsWith(META_DATA_VAR_PATTERN)) {
+                VariableDTO variableDTO = new VariableDTO(packVarInfo.getName(), "Global");
+                switch (packVarInfo.getType().getTag()) {
+                    case TypeTags.INT_TAG:
+                        variableDTO.setBValue(new BInteger(programFile.globalMemArea.getIntField(pkgIndex,
+                                packVarInfo.getGlobalMemIndex())));
+                        break;
+                    case TypeTags.BYTE_TAG:
+                        variableDTO.setBValue(new BByte((byte) (programFile.globalMemArea.getBooleanField(pkgIndex,
```
Comment (Kishanthan) [inline bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java]: Is this line formatted correctly ?

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java (hunk 2 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java]: +1. Addressed the suggestion.

File: bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java (hunk 3 above)
Comment (pranavan15) [reply bvm/ballerina-core/src/main/java/org/ballerinalang/util/debugger/Debugger.java]: Ack. Corrected.

**Pull request merged**